lab_2_tokenize_by_bpe/main.py

```diff
@@ -58,6 +106,27 @@ def train(
     :param num_merges: required number of new tokens
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
+    if (not (isinstance(word_frequencies, dict) or word_frequencies is None) or not
+        isinstance(num_merges, int)):
+        return None
+    merges_count = 0
+    while merges_count < num_merges:
+        token_pairs = count_tokens_pairs(word_frequencies)
+        if token_pairs is None:
+            return None
+        if token_pairs != {}:
+            new_dict = dict(
+                sorted(token_pairs.items(),
+                       key=lambda item: (-item[1], -len("".join(item[0])), "".join(item[0]))))
+            for k in new_dict.keys():
```

Comment:
What about `num_merges`? What if we only needed to merge, like, three pairs, and here you are merging everything...

Reply:
fixed